### PR TITLE
Stop and Start arcgis services around copy

### PR DESCRIPTION
**Pull request by steveoh**
refs #45

We do need to shut down services in arcgis server. This test proves that. 

## Setup

1. set `copyDestinations` to somewhere eg `c:\mapdata\`
1. copy the `NoSchemaLock.gdb` and `SchemaLock.gdb` to `c:\mapdata\`
1. open the two mxd's in `tests/maps` and change the data source to be `c:\mapdata\`
1. publish the two mxd's in `tests/maps`
  - forklift/NoSchemaLock
  - forklift/SchemaLock
1.  go in to NoSchemaLock and uncheck the schema lock box
1. save

## Test
1. Query the rest endpoint of the services. **make sure caching is disabled.** You should notice 1 feature is returned with the value of `Original`.
   - ```http://localhost:6080/arcgis/rest/services/forklift/NoSchemaLock/MapServer/0/query?where=1%3D1&f=pjson```
   - ```http://localhost:6080/arcgis/rest/services/forklift/SchemaLock/MapServer/0/query?where=1%3D1&f=pjson```

1. Run 🚜 on the `SchemaLockPallet.py` file in the tests folder. Your path may vary.
   - ``` python -m forklift lift ...\forklift\tests\SchemaLockPallet.py``
1. Query the rest endpoint of the services again. **make sure caching is disabled.** You should notice 2 features are returned with the value of `Original` and `New Data`.
   - ```http://localhost:6080/arcgis/rest/services/forklift/NoSchemaLock/MapServer/0/query?where=1%3D1&f=pjson```
   - ```http://localhost:6080/arcgis/rest/services/forklift/SchemaLock/MapServer/0/query?where=1%3D1&f=pjson```
1. Stop the services in arcgis server
1. Revert all the changes that were made to the gdb's. Then you can repeat the process again.

### CODE REVIEW

**Comment by stdavis** [comment]
@steveoh Is this ready for review/merge?

**Comment by steveoh** [comment]
Yup

File: src/forklift/lift.py [hunk 1]
```diff
@@ -9,8 +9,12 @@
 import logging
 import seat
 import shutil
+from arcgis import LightSwitch
 from arcpy import Compact_management, Describe
+from os import makedirs
 from os import path
+from os import remove
+from os import walk
```
Comment (stdavis) [inline src/forklift/lift.py]: Is it best practice to do `from ... import ...` imports on separate lines now?

File: src/forklift/secrets_sample.py [hunk 2]
```diff
@@ -9,3 +9,7 @@
 smtp_server = '<server>'
 smtp_port = '<port number>'
 from_address = '<email address>'
+
+ags_username = '<arcgis admin username>'
+ags_password = '<arcgis admin password>'
+ags_server = '<arcgis base url>'
```
Comment (stdavis) [inline src/forklift/secrets_sample.py]: `<arcgis base url>` should probably be changed to ip or domain name. Url made me think it should be something like: `http://localhost:6080/arcgis/manager`.

**Comment by stdavis** [comment]
Looks like `arcgis.py` could use some somes as well as `_copy_with_overwrite` in `test_lift.py`.

Otherwise it's looking good to me.

I don't think that it's worth changing at this point, but if we have problems in the future I think that the idea of shutting down one machine at a time may be an option.

File: src/forklift/lift.py (hunk 1 above)
Comment (steveoh) [inline src/forklift/lift.py]: I saw it somewhere and liked it. It's easier to remove them.

Comment (stdavis) [inline src/forklift/lift.py]: I can dig it. Same principle as
```
var x;
var y;
```
vs
```
var x, y;
```
in JS.

File: src/forklift/secrets_sample.py (hunk 2 above)
Comment (steveoh) [inline src/forklift/secrets_sample.py]: whats the thought on removing secretes in favor of environment variables?

Comment (stdavis) [inline src/forklift/secrets_sample.py]: I'm still up for that.

**Comment by stdavis** [comment]
Merged.

**Pull request closed without merging**